tests/unit/package_managers/bundler/test_parser.py

```diff
@@ -0,0 +1,327 @@
+def test_parse_lockfile_invalid_format(
+    mock_run_cmd: mock.MagicMock,
+    lockfile_content: str,
+    expected_error: str,
+    empty_gemfile: RootedPath,
+    rooted_tmp_path: RootedPath,
+) -> None:
+    lockfile_path = rooted_tmp_path.join_within_root(GEMFILE_LOCK)
+    lockfile_path.path.write_text(lockfile_content)
+
+    empty_json = json.loads(EMPTY_JSON)
+
+    if lockfile_content == LOCKFILE_INVALID_URL:
+        empty_json["dependencies"] = [
+            {
+                "name": "json-schema",
+                "version": "1.0.0",
+                "type": "git",
+                "url": "github",
+                "revision": "26487618a68443e94d623bb585cb464b07d36702",
+            }
+        ]
+    elif lockfile_content == LOCKFILE_INVALID_URL_SCHEME:
+        empty_json["dependencies"] = [
+            {
+                "name": "json-schema",
+                "version": "1.0.0",
+                "type": "git",
+                "url": "http://github.com/3scale/json-schema.git",
+                "revision": "26487618a68443e94d623bb585cb464b07d36702",
+            }
+        ]
+    elif lockfile_content == LOCKFILE_INVALID_REVISION:
+        empty_json["dependencies"] = [
+            {
+                "name": "json-schema",
+                "version": "1.0.0",
+                "type": "git",
+                "url": "https://github.com/3scale/json-schema.git",
+                "revision": "abcd1234",
+            }
+        ]
+    elif lockfile_content == LOCKFILE_INVALID_PATH:
+        empty_json["dependencies"] = [
+            {
+                "name": "pathgem",
+                "version": "1.0.0",
+                "type": "path",
+                "path": "/root/pathgem",
+            }
+        ]
+
+    mock_run_cmd.return_value = json.dumps(empty_json)
+
+    with pytest.raises((pydantic.ValidationError, UnexpectedFormat)) as exc_info:
+        parse_lockfile(rooted_tmp_path)
+
+    assert expected_error in str(exc_info.value)
+
+
+LOCKFILE_VALID = textwrap.dedent(
+    """
+    GIT
+      remote: https://github.com/3scale/json-schema.git
+      revision: 26487618a68443e94d623bb585cb464b07d36702
+      branch: master
+      specs:
+        json-schema (1.0.0)
+
+    PATH
+      remote: subpath/pathgem
+      specs:
+        pathgem (2.0.0)
+
+    GEM
+      remote: https://rubygems.org/
+      specs:
+        gem (3.0.0)
+
+    PLATFORMS
+      ruby
+      x86_64-linux
+
+    DEPENDENCIES
+      json-schema!
+      pathgem!
+      gem
+
+    BUNDLED WITH
+      2.5.10
+    """
+)
+
+
+@mock.patch("cachi2.core.package_managers.bundler.parser.run_cmd")
+def test_parse_gemlock(
+    mock_run_cmd: mock.MagicMock,
+    empty_gemfile: RootedPath,
+    rooted_tmp_path: RootedPath,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    lockfile_path = rooted_tmp_path.join_within_root(GEMFILE_LOCK)
+    lockfile_path.path.write_text(LOCKFILE_VALID)
+
+    empty_json = json.loads(EMPTY_JSON)
+
+    empty_json["dependencies"] = [
```

Comment:
An absolutely non-blocking nitpick: technically it is not empty starting from this line on.